### PR TITLE
Add Rio Salado College scraper

### PR DESCRIPTION
**Pull request by jodizzle**
This site is largely ASP.NET javascript shenanigans.  I was able to get it to work, but the result is unfortunately a bit messy.

### CODE REVIEW

File: osp_scraper/spiders/riosalado.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        terms in-between.
+
+        Currently, only two terms are available on the site, but this could
+        change in the future.
+        """
+        subj_url = "http://www.riosalado.edu/WebServices/Pages/schedule/sch.aspx?prefix={0}&term1={1}&term2={2}"
+        subjects = json.loads(response.body_as_unicode())
+        term1 = response.meta['terms'][0]
+        term2 = response.meta['terms'][-1]
+        for subj in subjects[:1]:
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/riosalado.py]: Is the `[:1]` supposed to be here?

File: osp_scraper/spiders/riosalado.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            )
+
+    def parse_for_syllabi(self, response):
+        courses = json.loads(response.body_as_unicode())
+        for course in courses:
+            anchor = " ".join([
+                "Term " + course['Term'],
+                "Section " + course['Sect'],
+                course['Course'],
+                course['Title'],
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/riosalado.py]: Minor: trailing comma.

**Comment by HarrisonGregg** [comment]
I notice that some of the syllabus URLs lead to a page like [this](https://matrix.riosalado.edu/RioWebApps/Public/Rio.SyllabusDefinition.UI/MiniSyllabusRender.aspx?courseId=ACC105&specialtyCode=&instructionMode=IN&startDate=05-01-2017&termCode=4174%C2%A7ionNum=12502).  Do you want to detect those and prevent them from being saved?  I think we've talked about it before and mostly decided against it, so maybe just add a comment about it?

File: osp_scraper/spiders/riosalado.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        Currently, only two terms are available on the site, but this could
+        change in the future.
+        """
+        subj_url = "http://www.riosalado.edu/WebServices/Pages/schedule/sch.aspx?prefix={0}&term1={1}&term2={2}"
+        subjects = json.loads(response.body_as_unicode())
+        term1 = response.meta['terms'][0]
+        term2 = response.meta['terms'][-1]
+        for subj in subjects[:1]:
+            yield scrapy.Request(
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/riosalado.py]: Is there a reason you're not using a `FormRequest` here?

File: osp_scraper/spiders/riosalado.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        match a given 'prefix' and occured during 'term1', 'term2' and/or all
+        terms in-between.
+
+        Currently, only two terms are available on the site, but this could
+        change in the future.
+        """
+        subj_url = "http://www.riosalado.edu/WebServices/Pages/schedule/sch.aspx?prefix={0}&term1={1}&term2={2}"
+        subjects = json.loads(response.body_as_unicode())
+        term1 = response.meta['terms'][0]
+        term2 = response.meta['terms'][-1]
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/riosalado.py]: Alternatively, it looks like you can just use `term1 = 1000` and `term2 = 9999`.  As it is, that seems to get a few courses for 2018, but I'm not sure if any of them have syllabi.

File: osp_scraper/spiders/riosalado.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+from ..spiders.CustomSpider import CustomSpider
+
+class RiosaladoSpider(CustomSpider):
+    name = "riosalado"
+
+    start_urls = ["http://www.riosalado.edu/Pages/leftnav/schedule.txt"]
+
+    def parse(self, response):
+        terms = sorted(list(filter(None, response.css("#terms option::attr(value)").getall())))
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/riosalado.py]: Do you need to include `list`?  Seems like sorted returns a list anyways.

File: osp_scraper/spiders/riosalado.py (hunk 1 above)
Comment (jodizzle) [inline osp_scraper/spiders/riosalado.py]: Nope, must be leftover test code.

**Comment by jodizzle** [comment]
@HarrisonGregg I think I addressed the majority of your comments.

As for detecting empty pages, when we've run into those situations in the past we haven't done any extra filtering unless it was convenient and could save us a request.  Does that approach make sense?  Doing extra filtering could save us downloads, but processing in that way seems like more of a job for post-scrape.

As for the terms trick, I think I'd rather deal with what the website gives us, even though my approach places a bit more trust in the website (more pieces functioning as intending) than yours does.  Unless you have strong objections?

**Pull request merged**